iosHyperskillApp/iosHyperskillApp/Sources/Modules/Track/Views/DiscoverNext/TopicToDiscoverNextButtonView.swift

```diff
@@ -49,24 +56,61 @@ struct TopicToDiscoverNextButtonView: View {
                                 .font(.subheadline)
                                 .foregroundColor(Color(ColorPalette.secondary))
                         }
+
+                        if isLearnNext {
+                            Image(Images.Home.ProblemOfDay.arrowUncompleted)
+                                .renderingMode(.original)
+                                .resizable()
+                                .frame(widthHeight: appearance.arrowIconSize)
+                        }
                     }
                 }
                 .padding(.horizontal)
                 .frame(minHeight: appearance.buttonStyle.minHeight)
                 .background(
-                    GeometryReader { geometry in
-                        Rectangle()
-                            .stroke(lineWidth: 0)
-                            .background(appearance.backgroundProgressColor)
-                            .cornerRadius(appearance.buttonStyle.cornerRadius)
-                            .frame(
-                                width: geometry.size.width * CGFloat(topic.progress?.completenessPercentage ?? 0) / 100
-                            )
-                    }
+                    buildCompletenessBar(
+                        completenessPercentage:
+                            topic.progress?.completenessPercentage ?? 0
+                    )
                 )
             }
         )
         .buttonStyle(appearance.buttonStyle)
+        .overlay(learnNextBadge)
```

Comment:
The "Learn next" badge does not animate on tap

Reply:
Right now we can not resolve this, please add `TODO`

Reply:
I tried to use the overlay in button label in order to apply animation to the badge but button label crops the badge so I decided to implement it in this way for now